### PR TITLE
Update webui.py

### PR DESCRIPTION
**Pull request by Schm1tz1**
Enabling authentication for local and shared UI (--listen and --share), see https://github.com/lllyasviel/Fooocus/issues/2126

### CODE REVIEW

**Comment by mashb1t** [comment]
Some other users and me have only recently chatted about this exact feature somewhere in the discussions 👍

**Comment by Schm1tz1** [comment]
NIce! I am using a reverse proxy to access Fooocus from the outside and are testing with Cloudflare Tunnels and I didn't want unauthenticated access so I tried this modification. Seems to be backwards-compatible, non-breaking.

File: webui.py [hunk 1]
```diff
@@ -618,6 +618,6 @@ def dump_default_english_config():
     server_name=args_manager.args.listen,
     server_port=args_manager.args.port,
     share=args_manager.args.share,
-    auth=check_auth if args_manager.args.share and auth_enabled else None,
+    auth=check_auth if auth_enabled else None,
```
Comment (mashb1t) [inline webui.py:621]:
```suggestion
    auth=check_auth if (args_manager.args.share or args_manager.args.listen) and auth_enabled else None,
```


I don't see any reason why we should only restrict this for users using --share and not also allow basic auth protection for --listen. For this you normally would also want to rather add args_manager.args.listen than also using basic auth locally. Is there any reason why you'd also propose to have basic auth locally?

**Review by mashb1t — CHANGES_REQUESTED** [review]

**Comment by mashb1t** [comment]
> NIce! I am using a reverse proxy to access Fooocus from the outside and are testing with Cloudflare Tunnels and I didn't want unauthenticated access so I tried this modification. Seems to be backwards-compatible, non-breaking.

IMHO the best case scenario in this case would be to not even route unauthenticated traffic to Fooocus at all but filter it in the reverse proxy.

File: webui.py (hunk 1 above)
Comment (Schm1tz1) [reply webui.py:621]: Fair point - didn't have the local mode on screen but why not allow authentication in general and let the user decide, that's what I did here. You still actively need to create the `auth.json` file so no surprises are to be expected.

**Comment by Schm1tz1** [comment]
> > NIce! I am using a reverse proxy to access Fooocus from the outside and are testing with Cloudflare Tunnels and I didn't want unauthenticated access so I tried this modification. Seems to be backwards-compatible, non-breaking.
> 
> IMHO the best case scenario in this case would be to not even route unauthenticated traffic to Fooocus at all but filter it in the reverse proxy.

For that case it makes senes but let's not discuss that here - it depends a lot on the whole setup and how your Ingress / Proxy / NAT setup looks like. It's just about allowing basic authentication on Fooocus - one can still decide to move it or not.

**Pull request merged**